scipy/optimize/tests/test_zeros.py

```diff
@@ -661,3 +661,24 @@ def fpp_array(x):
         f, x0_array, fprime=fp, fprime2=fpp_array, full_output=True
     )
     assert result.converged.all()
+
+
+def test_gh9254_flag_if_maxiter_exceeded():
+    """
+    Test that if the maximum iterations is exceeded that the flag is not
+    converged.
+    """
+    maximum_iterations = 10
+    r = zeros.brentq(
+        lambda x: ((1.2*x - 2.3)*x + 3.4)*x - 4.5,
+        -30, 30, (), 1e-6, 1e-6, maximum_iterations,
+        full_output=True, disp=False)
+    assert r[1].flag == zeros.CONVERR
+    assert r[1].iterations == maximum_iterations
+    more_maximum_iterations = 100
+    r = zeros.brentq(
+        lambda x: ((1.2*x - 2.3)*x + 3.4)*x - 4.5,
+        -30, 30, (), 1e-6, 1e-6, more_maximum_iterations,
+        full_output=True, disp=False)
```

Comment:
These `brentq` calls are duplicates except for the iteration counts. I haven't been "enforcing" parametrization with pytest too much though, so minor point.

Reply:
I'm +1 (modulo the style nit pick in the block setting the flag to CONVERGED.)